compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -452,7 +452,27 @@ impl<'tcx> WfPredicates<'tcx> {
                     match ct.kind() {
                         ty::ConstKind::Unevaluated(uv) => {
                             if !ct.has_escaping_bound_vars() {
-                                let obligations = self.nominal_obligations(uv.def.did, uv.substs);
+                                let substs = if let Some(did) = uv.def.did.as_local()
+                                    && self
+                                        .tcx
+                                        .hir()
+                                        .opt_const_param_default_param_def_id(
+                                            self.tcx.hir().local_def_id_to_hir_id(did),
+                                        )
+                                        .is_some()
+                                    && let Some(did) = self.tcx.opt_parent(uv.def.did)
+                                {
+                                    // HACK(TaKO8Ki) `WfPredicates::nominal_obligations` calls `SubstsFolder::fold_const` in it.
+                                    // If `uv.substs` is used, it will cause an ICE because `uv.substs` does not include the const.
+                                    self.tcx.mk_substs(
+                                        InternalSubsts::identity_for_item(self.tcx, did)
+                                            .into_iter()
+                                            .chain(uv.substs),
```

Comment:
This needs an explanation of what it's doing

Reply:
I've added a comment.

Reply:
Sorry, specifically I was asking about a comment explains why you're chaining the identity parent substs with the (partially constructed) substs you have in `uv.substs`. 

It doesn't seem correct to me at first glance.